Diz.Core/model/snes/Data.cs

```diff
@@ -0,0 +1,471 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Serialization;
+using Diz.Core.arch;
+using Diz.Core.export;
+using Diz.Core.model.byteSources;
+using Diz.Core.util;
+using JetBrains.Annotations;
+
+namespace Diz.Core.model.snes
+{
+    // old-style data container to phase out eventually.
+    // right now this supports just 1 ByteSource that is a SNES address space with an attached Rom
+    //
+    // the underlying new ByteSource stuff it uses supports more, but right now, the rest of the app doesn't.
+    // Data is the bridge between the old and new.
```

Comment:
the ```Data``` class was originally the main event, now it's mostly just the bridge between the newer data-holding interfaces (ByteSource and friends) and a backwards-compatible API that makes varying degrees of sense in the new world here.

There's a lot that can be pulled out of this class and cleaned up, probably a good place to spend some focus.